markdownlint.js

```diff
@@ -161,7 +161,37 @@ function printResult(lintResult) {
     process.exitCode = 1;
   }
 
-  if (program.output) {
+  if (program.junit) {
+    const builder = require('junit-report-builder');
+    const suite = builder
+      .testSuite()
+      .name('Markdownlint')
+      .timestamp(new Date().toISOString());
+    if (results.length > 0) {
+      results.forEach(result => {
+        const {file, lineNumber, column, names, description} = result;
+        const columnText = column ? `:${column}` : '';
+        suite
+          .testCase()
+          .className(file)
+          .name(`${file}:${lineNumber}${columnText} ${names}`)
+          .failure(`${file}:${lineNumber}${columnText} ${names} ${description}`, names)
+          .time(0);
+      });
+    } else {
+      suite.testCase().className(program.args).name('Markdownlint');
+    }
+
+    suite.time(0);
```

Comment:
Can you help me understand the difference between timestamp above and time(0) here and above?

Reply:
Sure. The JUnit format has a timestamp indicating when the Test Suite was started, and then Time indicates the time (in seconds) that the test suite took.  Test cases also have a time in second, but no timestamp to indicate when they started.  The fact that `testSuite.time(0)` is here is I think a relic of when I was trying to actually calculate the time it took to generate the report, so I couldn't set the time until the report was done.  However after looking at what eslint did (just setting the time to 0) I removed the calculation but left setting it there.  There's no reason why it can't be moved into the construction of the testSuite since I know the time will be 0.

But regardless of where it's defined, they're two separate values.  Eslint just doesn't set the timestamp at all, but it was one of the JUnit fields, separate from time.